independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/InterceptorInfo.java

```diff
@@ -57,6 +59,12 @@ public class InterceptorInfo extends BeanInfo implements Comparable<InterceptorI
                 if (Modifier.isStatic(method.flags())) {
                     continue;
                 }
+                if (method.hasAnnotation(DotNames.PRODUCES) || method.hasAnnotation(DotNames.DISPOSES)) {
```

Comment:
We should look into the `AnnotationStore` here, IMHO. Something like:

```java
if (Annotations.contains(beanDeployment.getAnnotations(method), DotNames.PRODUCES)
        || Annotations.contains(beanDeployment.getAnnotations(method), DotNames.DISPOSES)) {
    ...
}
```

Reply:
Though I can see we don't do this below either... which is probably wrong as well :-)